code/parachain/runtime/composable/src/xcmp.rs

```diff
@@ -333,25 +333,151 @@ impl<
 	}
 }
 
-pub struct ForeignXcm;
+pub struct ForeignAssetsToXcm;
 
-impl Convert<CurrencyId, Option<MultiLocation>> for ForeignXcm {
+impl Convert<CurrencyId, Option<MultiLocation>> for ForeignAssetsToXcm {
 	fn convert(a: CurrencyId) -> Option<MultiLocation> {
 		match AssetsRegistry::asset_to_remote(a) {
+			// XCMed assets
 			Some(ForeignAssetId::Xcm(VersionedMultiLocation::V3(xcm))) => Some(xcm),
+			// IBCed assets
+			Some(ForeignAssetId::IbcIcs20(_)) => Some(MultiLocation {
+				parents: 0,
+				interior: X3(
+					Parachain(ParachainInfo::parachain_id().into()),
+					PalletInstance(<AssetsRegistry as PalletInfoAccess>::index() as u8),
+					GeneralIndex(a.into()),
+				),
+			}),
 			_ => None,
 		}
 	}
 }
 
-impl Convert<MultiLocation, Option<CurrencyId>> for ForeignXcm {
+impl Convert<MultiLocation, Option<CurrencyId>> for ForeignAssetsToXcm {
 	fn convert(a: MultiLocation) -> Option<CurrencyId> {
 		AssetsRegistry::location_to_asset(ForeignAssetId::Xcm(VersionedMultiLocation::V3(a)))
 	}
 }
 
-type AssetsIdConverter =
-	CurrencyIdConvert<ForeignXcm, primitives::topology::Composable, ParachainInfo>;
+type AssetsIdConverter = CurrencyIdConvert<
+	ForeignAssetsToXcm,
+	primitives::topology::Composable,
+	AssetsRegistry,
+	ParachainInfo,
+>;
+
+#[cfg(test)]
+mod test {
```

Comment:
no important,

but mostly useless. it does not tests reality on xcm location inversion, like simulator did